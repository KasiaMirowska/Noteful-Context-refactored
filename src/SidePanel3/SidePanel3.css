.side-panel3 {
   text-align: right;
   
}
.panel3 {
    margin-right: 30px;
}


@media (min-width: 600px) {
    .side-panel3 {
        flex: 1;
        margin: 0;
    }
  }  