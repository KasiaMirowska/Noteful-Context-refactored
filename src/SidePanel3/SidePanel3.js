import React from 'react';
import PropTypes from 'prop-types';
import './SidePanel3.css';

export default function SidePanel3(props) {
        return (
            <div className='side-panel3'>
                <button className='panel3' type='button' onClick={() => props.history.goBack()}>Back</button>
              
            </div>
        )
    
}
SidePanel3.propTypes = {
    history: PropTypes.object.isRequired,

}