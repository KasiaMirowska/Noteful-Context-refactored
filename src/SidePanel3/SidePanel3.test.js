import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import SidePanel3 from './SidePanel3';
import { shallow } from 'enzyme'; 

describe('SidePanel3', () => {
    const props = {
        history: {
            goBack: jest.fn()
        },
        location: {
            pathname: '/newFolder'
        }
    }
    it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDom.render(<SidePanel3 {...props}/>, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const sidePanel3 = renderer.create(<SidePanel3 {...props}/>);

        expect(sidePanel3.toJSON()).toMatchSnapshot();
    });

    it('goes back on click', () => {
        const onClickFn = props.history.goBack;
        const wrapper = shallow(<SidePanel3 {...props}/>);
        wrapper.find('button').simulate('click');
        expect(onClickFn).toBeCalled();
    });
});
