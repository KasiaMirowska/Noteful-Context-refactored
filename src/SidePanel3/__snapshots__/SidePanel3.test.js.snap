// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel3 renders the UI as expected 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
</div>
`;
